src/fileformats/ocd_file_import.cpp

```diff
@@ -1525,30 +1525,51 @@ void OcdFileImport::setupLineSymbolPointSymbols(OcdFileImport::OcdImportedLineSy
 	}
 }
 
-void OcdFileImport::mergeLineSymbol(CombinedSymbol* full_line, LineSymbol* main_line, LineSymbol* framing_line, LineSymbol* double_line)
+bool OcdFileImport::hasSecondarySymbol(const Ocd::LineSymbolCommonV8& attributes) const
+{
+	return attributes.secondary_data_size > 0;
+}
+
+void OcdFileImport::setupSecondarySymbol(OcdImportedLineSymbol* line_symbol, const Ocd::LineSymbolCommonV8& attributes, const Ocd::PointSymbolElementV8* elements)
 {
-	full_line->setNumParts(3); // reserve
+	importLineSymbolBase(line_symbol, attributes);
+	line_symbol->setColor(nullptr);
+	line_symbol->setLineWidth(0);
 	
-	int part = 0;
-	full_line->setPart(part++, main_line, true);
-	main_line->setHidden(false);
-	main_line->setProtected(false);
-	main_line->setName(main_line->getName() + tr(" - main line"));
+	auto* point_symbol = new OcdImportedPointSymbol();
+	const Ocd::OcdPoint32* coords = reinterpret_cast<const Ocd::OcdPoint32*>(elements) + attributes.primary_data_size;
```

Comment:
Oh. Do we check boundaries, in similar contexts?

Reply:
We discussed that in an OpenOrienteering "team internal" discussion back in August 2018. The conclusion was that I don't mind crashes resulting from invalid input, while you promoted more convenient behavior in the face of invalid data. 

I understand your point, but it adds error handling code. From my POV, as long as we don't de-reference such a pointer for **write**, we should be shielded from the worst security issues.